### PR TITLE
Fixed some deprecations

### PR DESCRIPTION
**Pull request by artursvonda**
Fix some of the easier deprecated calls and route path.

### CODE REVIEW

File: Validator/LocaleValidator.php [hunk 1]
```diff
@@ -79,9 +79,9 @@ public function validate($locale, Constraint $constraint)
         $locale = (string) $locale;
 
         if ($this->intlExtension) {
-            $primary = SymfonyLocale::getPrimaryLanguage($locale);
-            $region  = SymfonyLocale::getRegion($locale);
-            $locales = SymfonyLocale::getLocales();
+            $primary = \Locale::getPrimaryLanguage($locale);
+            $region  = \Locale::getRegion($locale);
```
Comment (dbu) [inline Validator/LocaleValidator.php]: that looks wrong

**Comment by dbu** [comment]
i guess its time to drop symfony 2.2 support. @lunetics do you agree? we should not bother with BC for 2.2 anymore, its so much end of life.

**Comment by artursvonda** [comment]
Was thinking the same thing. I actually updated travis file for that and added Symfony 2.8. Should I push it as part of this PR?

File: Validator/LocaleValidator.php (hunk 1 above)
Comment (artursvonda) [inline Validator/LocaleValidator.php]: Felt weird too but that looks like the way to go. \Locale is part of ext-intl and Symfony Intl Component provides stubs if ext-intl is not loaded.

**Comment by dbu** [comment]
lets remove 2.2 in this PR and create a separate one for 2.8. 2.8 should also be in allowed failures for now, there might be deprecations and other issues. 2.7 however should no longer be allowed to fail, its released.

**Comment by dbu** [comment]
@lunetics can you have a look and decide if we can merge? also once this is merged would be a good moment to tag a release (see #146)

**Comment by lunetics** [comment]
yes dropping 2.2 support is fine :) merged

**Pull request merged**